File: src/test/java/io/vertx/support/SubHandler.java
package io.vertx.support;

import io.vertx.codegen.annotations.VertxGen;
import io.vertx.codetrans.MethodExpressionTest;
import io.vertx.core.Handler;

/**
 * @author <a href="mailto:julien@julienviet.com">Julien Viet</a>
 */
@VertxGen
public interface SubHandler extends Handler<String> {

  static void classHandler(Handler<String> handler) {
    handler.handle("hello_class");
  }

  static SubHandler create() {
    return event -> {
      MethodExpressionTest.event = event;
    };
  }

  default void instanceHandler(Handler<String> handler) {
    handler.handle("hello_instance");
  }
}

File: src/test/scala/io/vertx/scala/support/SubHandler.scala
package io.vertx.scala.support

import io.vertx.codetrans.MethodExpressionTest
import io.vertx.core.Handler

/**
 * @author <a href="mailto:jochen.mader@codecentric.de">Jochen Mader</a>
 */

class SubHandler extends Handler[String] {
  def handle(event: String):Unit = {
    MethodExpressionTest.event = event
  }

  def instanceHandler(handler: Handler[String]):Unit = {
    handler.handle("hello_instance")
  }
}

object SubHandler {
  def create() = {
    new SubHandler()
  }
  def classHandler(handler: Handler[String]):Unit = {
    handler.handle("hello_class")
  }
}
